packages/db/index.js

```diff
@@ -160,20 +160,23 @@ export class DBClient {
   }
 
   /**
-   * Get used storage in bytes.
+   * Get used storage in bytes, both uploaded and pinned.
    *
    * @param {number} userId
-   * @returns {Promise<number>}
+   * @returns {Promise<import('./db-client-types').UsedStorage>}
    */
   async getUsedStorage (userId) {
-    /** @type {{ data: string, error: PostgrestError }} */
+    /** @type {{ data: { uploaded: string, pinned: string }, error: PostgrestError }} */
     const { data, error } = await this._client.rpc('user_used_storage', { query_user_id: userId }).single()
 
     if (error) {
       throw new DBError(error)
     }
 
-    return data || 0 // No uploads for the user
+    return {
+      uploaded: parseTextToNumber(data.uploaded),
```

Comment:
The `MAX_SAFE_INTEGER` is `9007199254740991`,  if I've done my conversion right that translates to  `7.999999999999999112 Pebibyte`, which for a single user should be safe enough? 
And even if not safe we're raising if that's the case.
I wonder if letting the client deal with it is just moving a problem that we'd have anyway?

Reply:
> which for a single user should be safe enough?

I think it should be enough yes, even though I feel it would be safer to simply rely on BigInt in client instead.

> I wonder if letting the client deal with it is just moving a problem that we'd have anyway?

The client would be able to use BigInt. Otherwise we would need to serialize from API to client.

What do you think?

Reply:
I think both solutions have their merits and cons, and I'm not too opinionated to go one route or the other.

Here's main drivers for having the DB package accountable for the conversion (and possible error) has the merits of:
- db client packages don't need to be aware of the casting (and its problems). That means if we end up having to use that number in other contexts is less likely we end up with a "transparent" bugs.
- The same applies to APIs clients.

I'd like to avoid that, those are the kind of bugs that are really tricky to find.

Reply:
it's not a big deal for now. If we are lucky enough to get a user with 7 PiB, then we can review this. For now, let's do the most convenient thing for the calling code and give them numbers as numbers. If this turns out to be a problem, then we can send them as strings and let the rendering code parse it with BigInt, which is well supported now.